numpy/core/tests/test_umath.py

```diff
@@ -1808,6 +1808,44 @@ def test_special(self):
             assert_equal(ncu.log1p(-2.), np.nan)
             assert_equal(ncu.log1p(-np.inf), np.nan)
 
+class TestLog1pComplex:
+    def test_log1p_complex(self):
+        assert_almost_equal(ncu.log1p(0.2 + 0.3j), ncu.log(1.2 + 0.3j))
+        assert_almost_equal(ncu.log1p(1e-19 + 1e-18j), 1e-19 + 1e-18j,
+                            decimal=23)
```

Comment:
I suspect that `assert_almost_equal_nulp` is the more convenient way of checking things here.

@WarrenWeckesser if you could want to dig into whether the implementation looks right, that would be great.  I would still feel things would be much simpler if we can "steal" an implementation from elsewhere (that has a clearly BSD compatible license).

Reply:
Just to centralise ideas (https://github.com/pytorch/pytorch/pull/89214#issuecomment-1319832800), I agree. Now, it's not clear where is this function implemented for complex numbers really...

Reply:
Done